### PR TITLE
Cognito-IDP describe_resource_server() and list_resource_servers()

### PR DESCRIPTION
**Pull request by pll**
In `moto/cognitoidp/models.py`, added:
- `describe_resource_server()`
- `list_resource_servers()`

In `moto/cognitoidp/responses.py`, added:
- `describe_resource_server()`
- `list_resource_servers()`

In `moto/cognitoidp/utils.py`, added:
- `list_resource_servers` to `PAGINATION_MODEL`

In `tests/test_cognitoidp/test_cognitoidp.py`:
- changed `test_resource_server()` to `test_create_resource_server()`
- added `test_describe_resource_server()`
- added `test_list_resource_servers_empty_set()`
- added `test_list_resource_servers_single_page()`
- added `test_list_resource_servers_multi_page()`

Additionally:
- Ran `update_backend_index.py`
- Ran `make format`
- Ran `make lint`

### CODE REVIEW

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/cognitoidp/responses.py [hunk 1]
```diff
@@ -534,6 +534,32 @@ def create_resource_server(self) -> str:
...
+            user_pool_id, identifier
+        )
+        return json.dumps({"ResourceServer": resource_server.to_json()})
+
+    def list_resource_servers(self) -> str:
+        max_results = self._get_param("MaxResults")
+        next_token = self._get_param("NextToken")
+        user_pool_id = self._get_param("UserPoolId")
+        # import pytest
+        # pytest.set_trace()
```
[the remark this answers is not in the document]
Comment (pll) [reply moto/cognitoidp/responses.py]: @bblommers  - Not nitpicky at all! I should have scanned my code better :-/

Removed and pushed.

**Comment by pll** [comment]
@bblommers resolved.

**Review by bblommers — APPROVED** [review]
LGTM - thank you for adding these features @pll!

**Comment by pll** [comment]
@bblommers - My pleasure, always happy to contribute to something I depend on!  I'm sure I'll be back again ;)

**Pull request merged**